tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -850,5 +850,523 @@ int main(int argc, char **argv)
+
+                /* Sanity check: non-chacha20 RSA ciphersuite chosen without chacha20 boosting enabled */
+                /* cppcheck-suppress redundantAssignment */
+                cipher_preferences.allow_chacha20_boosting = false;
+                EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(connection, dhe_chacha20_boosted_wire, count));
+                EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_rsa_with_aes_256_gcm_sha384);
+            }
+
+            /* Server is able to negotiate its most preferred chacha20 ciphersuite */
+            {
+                DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+                EXPECT_NOT_NULL(connection);
+                connection->security_policy_override = &security_policy;
+                connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(connection, S2N_TLS12));
+                EXPECT_SUCCESS(s2n_connection_set_config(connection, server_config));
+
+                static struct s2n_cipher_suite *test_cipher_suite_list[] = {
+                    /* Skipped because it is not a chacha20 ciphersuite */
+                    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       
+                    /* Skipped because it is not a chacha20 ciphersuite. Is negotiated if chacha20 boosting is disabled.  */
+                    &s2n_ecdhe_rsa_with_aes_256_cbc_sha384,         
+                    /* First chacha20 ciphersuite and is negotiated */
+                    &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, 
+                    /* Second chacha20 ciphersuite and is negotiated */
+                    &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   
+                };
+
+                cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(test_cipher_suite_list),
+                    .suites = test_cipher_suite_list,
+                    .allow_chacha20_boosting = true
+                };
+
+                /*
+                 * Client signalled chacha20 boosting.
+                 * if chacha20_boosting is enabled by the server:
+                 *      TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 > TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 
+                 * else:
+                 *      TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 > TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 
+                 */
+                uint8_t ecdhe_chacha20_boosted_wire[] = {
+                    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384,
+                };
+                uint8_t count = sizeof(ecdhe_chacha20_boosted_wire) / S2N_TLS_CIPHER_SUITE_LEN;
+
+                /* Verify ecdhe ecdsa chacha20 ciphersuite chosen with chacha20 boosting enabled */
```

Comment:
The "ecdhe ecdsa" part is irrelevant to this test. You're just looking for the most preferred ciphersuite. Same for the mention of "aes_256_cbc" below-- not relevant to the test. Focus on what's important to the test.